test/fuzz/parser_fuzzers.go

```diff
@@ -0,0 +1,337 @@
+	"k8s.io/kubernetes/pkg/kubelet/eviction"
+	"k8s.io/kubernetes/pkg/registry/core/service/portallocator"
+	"k8s.io/kubernetes/pkg/util/parsers"
+
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
+)
+
+// FuzzParseSelector implements a fuzzer
+// that targets fields.ParseSelector
+func FuzzParseSelector(data []byte) int {
+	_, _ = fields.ParseSelector(string(data))
+	return 1
+}
+
+// FuzzParseQuantity implements a fuzzer
+// that targets resource.ParseQuantity
+func FuzzParseQuantity(data []byte) int {
+	_, _ = resource.ParseQuantity(string(data))
+	return 1
+}
+
+// FuzzParseGroupVersion implements a fuzzer
+// that targets schema.ParseGroupVersion
+func FuzzParseGroupVersion(data []byte) int {
+	_, _ = schema.ParseGroupVersion(string(data))
+	return 1
+}
+
+// FuzzParseResourceArg implements a fuzzer
+// that targets schema.ParseResourceArg
+func FuzzParseResourceArg(data []byte) int {
+	_, _ = schema.ParseResourceArg(string(data))
+	return 1
+}
+
+// FuzzLabelsParse implements a fuzzer
+// that targets labels.Parse
+func FuzzLabelsParse(data []byte) int {
+	_, _ = labels.Parse(string(data))
+	return 1
+}
+
+// FuzzParseImageName implements a fuzzer
+// that targets parsers.ParseImageName
+func FuzzParseImageName(data []byte) int {
+	_, _, _, _ = parsers.ParseImageName(string(data))
+	return 1
+}
+
+// FuzzParsePrivateKeyPEM implements a fuzzer
+// that targets keyutil.ParsePrivateKeyPEM
+func FuzzParsePrivateKeyPEM(data []byte) int {
+	_, _ = keyutil.ParsePrivateKeyPEM(data)
+	return 1
+}
+
+// FuzzParsePublicKeysPEM implements a fuzzer
+// that targets keyutil.ParsePublicKeysPEM
+func FuzzParsePublicKeysPEM(data []byte) int {
+	_, _ = keyutil.ParsePublicKeysPEM(data)
+	return 1
+}
+
+// FuzzParseCSR implements a fuzzer
+// that targets certificate.ParseCSR
+func FuzzParseCSR(data []byte) int {
+	_, _ = certificate.ParseCSR(data)
+	return 1
+}
+
+// FuzzMeta1ParseToLabelSelector implements a fuzzer
+// that targets metav1.ParseToLabelSelector
+func FuzzMeta1ParseToLabelSelector(data []byte) int {
+	_, _ = metav1.ParseToLabelSelector(string(data))
+	return 1
+}
+
+// FuzzParseHostPort implements a fuzzer
+// that targets kubeadmutil.ParseHostPort
+func FuzzParseHostPort(data []byte) int {
+	_, _, _ = kubeadmutil.ParseHostPort(string(data))
+	return 1
+}
+
+// FuzzParseVersion implements a fuzzer
+// that targets:
+// - version.ParseSemantic,
+// - version/(*Version).String()
+// - version.ParseGeneric
+// - version/(*Version).AtLeast(*Version)
+func FuzzParseVersion(data []byte) int {
+	f := fuzz.NewConsumer(data)
+	vString1, err := f.GetString()
+	if err != nil {
+		return 0
+	}
+	v1, err := version.ParseSemantic(vString1)
+	if err != nil {
+		return 0
+	}
+
+	// Test if the Version will crash (*Version).String()
+	_ = v1.String()
+
+	vString2, err := f.GetString()
+	if err != nil {
+		return 0
+	}
+	v2, err := version.ParseGeneric(vString2)
+	if err != nil {
+		return 0
+	}
+	_ = v1.AtLeast(v2)
+	return 1
```

Comment:
Why isn't this a separate fuzzer?

Reply:
This fuzzer is really targeting `version.parse()` which will be fully covered by calling `version.ParseGeneric` and `version.ParseSemantic` instead of calling `version.parse(string(data), false)` and calling `version.parse(string(data), true)`. 
The reason `version.AtLeast` is included in the fuzzer is that the parsers are returning `*version.Version`s which makes it possible to test `version.checkInternal` as well by calling `version.AtLeast`